bin/pycbc_source_probability_offline

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+"""
+Compute source probabilities using mchirp estimation method for all events
+in a chunk with an IFAR above certain threshold.
+"""
+import os
+import argparse
+import h5py, logging
+import numpy as np
+import json
+import tqdm
+from pycbc.io import hdf
+from pycbc.pnutils import mass1_mass2_to_mchirp_eta
+from pycbc import mchirp_area
+import pycbc.version
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--trigger-file')
+parser.add_argument('--bank-file')
+parser.add_argument('--single-detector-triggers', nargs='+', default=None)
+parser.add_argument('--search-tag',
+                    help='eg: PYCBC_AllSky, PYCBC_HighMass')
+parser.add_argument('--ifar-threshold', type=float, default=None,
+                    help='Select only candidate events with IFAR '
+                         'above threshold.')
+parser.add_argument('--verbose', action='count')
+parser.add_argument("--version", action="version",
+                    version=pycbc.version.git_verbose_msg)
+mchirp_area.insert_args(parser)
+args = parser.parse_args()
+
+mc_area_args = mchirp_area.from_cli(args)
+
+if args.verbose:
+    log_level = logging.INFO
+    logging.basicConfig(format='%(asctime)s : %(message)s', level=log_level)
+
+logging.info('Using files: %s, %s, %s' % 
+             ((args.trigger_file).split('/')[-1],
+              (args.bank_file).split('/')[-1],
+              (',').join([name.split('/')[-1] for name
+                          in args.single_detector_triggers])))
+
+GPS_START_TIME = args.trigger_file.split('-')[2]
+GPS_START_TIME_NS = args.trigger_file.split('-')[3].split('.')[0]
+
+dir_path = 'source_probability_results/CHUNK_%s-%s/' % (GPS_START_TIME,
+                                                        GPS_START_TIME_NS)
+if not os.path.exists(dir_path):
+    try:
+        os.makedirs(dir_path)
+        logging.info('Created directory %s' % dir_path)
+    except OSError:
```

Comment:
I'm not sure why you need to have a specially coded try-except switch here - what is the problem if you just write os.makedirs and if there is some sort of error the program should stop and report the error anyway